stylesheets/_component.buttons.scss

```diff
@@ -40,222 +48,251 @@
+            &.active {
+                background-color: transparent !important;
+                color: color($state, dark) !important;
+            }
+        }
     }
-}
 
-@mixin btn-active($base) {
-    &.active {
-    $border-active: adjust-color($base, $saturation: 9%, $lightness: -14%);
-    $inset-shadow-active: adjust-color($base, $saturation: 7%, $lightness: -17%);
-    border: 1px solid $border-active;
-    box-shadow: inset 0 0 8px 4px $inset-shadow-active, inset 0 0 8px 4px $inset-shadow-active, 0 1px 1px 0 #eee;
+    > .badge {
+        background-color: color($state, dark);
+        color: pick_best_color(color($state, dark), (#fff, color(gray, dark)));
+        text-shadow: 0 -1px 0 rgba(0, 0, 0, .25);
     }
-}
 
-.btn {
-    @include button-states('base');
-}
+    &.btn--naked {
+        color: color($state, dark) !important;
 
-.btn--primary {
-    @include button-states('primary');
+        &:active,
+        &.active {
+            background-color: color($state, light) !important;
+            color: color($state, dark) !important;
 
-    .badge {
-        background-color: $badge-primary-bg-darker;
-        text-shadow: 0 -1px 0 rgba(0, 0, 0, .25);
+            &:not(:disabled):not(.is-disabled).btn--white {
+                color: color(text) !important;
+            }
+        }
     }
 }
 
-.btn--success {
-    @include button-states('success');
+/**
+ * UI states
+ */
+
+.btn {
+    @include button-states('base');
+    border-bottom-style: solid;
+    text-decoration: none;
 
-    .badge {
-        background-color: $badge-success-bg-darker;
-        text-shadow: 0 -1px 0 rgba(0, 0, 0, .25);
+    &:disabled,
+    &.is-disabled,
+    &.disabled {
+        cursor: not-allowed !important;
     }
 }
 
-.btn--warning {
-    @include button-states('warning');
-
-    .badge {
-        background-color: $badge-warning-bg-darker;
-        text-shadow: 0 -1px 0 rgba(0, 0, 0, .25);
+@each $state, $state-color, $state-color-alt in $state-colors {
+    &.btn--#{$state} {
+        @include button-states($state);
     }
 }
 
-.btn--danger {
-    @include button-states('danger');
+.btn--outline {
+    background: none !important;
+    border: 0;
+    box-shadow: inset 0 0 0 2px color(base);
+    padding-bottom: $btn-padding-bottom + 2;
+    padding-top: $btn-padding-top;
 
-    .badge {
-        background-color: $badge-danger-bg-darker;
-        text-shadow: 0 -1px 0 rgba(0, 0, 0, .25);
+    &:hover,
+    &.is-hovered {
+        box-shadow: inset 0 0 0 2px color(base);
+        padding-bottom: $btn-padding-bottom + 1;
     }
-}
-
-.btn--info {
-    @include button-states('info');
 
-    .badge {
-        background-color: $badge-info-bg-darker;
-        text-shadow: 0 -1px 0 rgba(0, 0, 0, .25);
+    &:active,
+    &.active {
+        background-color: color(base, light) !important;
+        box-shadow: inset 0 0 0 2px color(base);
+        color: pick_best_color(color(base, light), (#fff, color(gray, dark))) !important;
```

Comment:
Are there too many parentheses here? Highlighting looks odd in the diff where the `#fff, color(gray, dark)` is wrapped, so just making sure it's intentional.